apps/distgo/cmd/dist/dist.go

```diff
@@ -30,11 +30,48 @@ import (
 	"github.com/palantir/godel/apps/distgo/cmd"
 	"github.com/palantir/godel/apps/distgo/cmd/build"
 	"github.com/palantir/godel/apps/distgo/params"
+	"github.com/palantir/godel/apps/distgo/pkg/imports"
 	"github.com/palantir/godel/apps/distgo/pkg/osarch"
 	"github.com/palantir/godel/apps/distgo/pkg/script"
 	"github.com/palantir/godel/apps/distgo/pkg/slsspec"
 )
 
+func RequiresDist(products []string, cfg params.Project, wd string) ([]string, error) {
+	buildSpecsWithDeps, err := build.SpecsWithDepsForArgs(cfg, products, wd)
+	if err != nil {
+		return nil, err
+	}
+	productsSlice := []string{}
+	productsMap := make(map[string]struct{})
+	for _, curSpecWithDeps := range buildSpecsWithDeps {
+		for _, curDistCfg := range curSpecWithDeps.Spec.Dist {
+			artifactPaths := FullArtifactsPaths(ToDister(curDistCfg.Info), curSpecWithDeps.Spec, curDistCfg)
+			for _, artifactPath := range artifactPaths {
+				if fi, err := os.Stat(artifactPath); err == nil {
```

Comment:
we cannot directly continue. The error not being nil means that the artifact does not exist at the destination path and you need to build it. So, we need to add it to the productsMap.